brave/brave-core
# [Tab Switch]: Prefer switching to an existing tab over navigating to the same url in a new one

Resolves https://github.com/brave/brave-browser/issues/32896

Without flag:
![image](https://github.com/brave/brave-core/assets/7678024/840fca27-b795-4736-a724-35db0e05d0a4)

With flag:
![image](https://github.com/brave/brave-core/assets/7678024/a6501893-1779-479c-bda2-f09314397f7f)

The plan is to enable this in Nightly for a while (via Griffin) and see what people think



## Submitter Checklist:

  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`

## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:

1. Enable brave://flags/#brave-omnibox-tab-switch-by-default
2. Enter a currently open tab (for some reason, not all open tabs show the switcher, I'm not sure why, but it comes from upstream)
3. The URL bar should switch you to the existing tab by default, rather than forcing you to tab
